app/models/bulk_upload.rb

```diff
@@ -65,9 +65,8 @@ def save_attachments
   end
 
   def attachments_must_be_valid
-    unless attachments.all? { |a| a.valid?(context: :user_input) }
-      errors.add(:base, message: "Please enter missing fields for each attachment")
-    end
+    attachments.each { |a| a.valid?(context: :user_input) }
+    errors.add(:base, message: "Please enter missing fields for each attachment") unless attachments.map(&:errors).all?(&:blank?)
```

Comment:
This might be a more idiomatic way of writing it:

```suggestion
    invalid_attachments = attachments.reject { |a| a.valid?(context: :user_input) }
    errors.add(:base, message: "Please enter missing fields for each attachment") if invalid_attachments.present?
```